pkgs/development/python-modules/basicsr/default.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+
+# build-system
+, cython
+, setuptools
+
+# dependencies
+, addict
+, future
+, lmdb
+, numpy
+, opencv4
+, pillow
+, pyyaml
+, requests
+, scikit-image
+, scipy
+, tensorboard
+, torch
+, torchvision
+, tqdm
+, yapf
+}:
+
+buildPythonPackage rec {
+  pname = "basicsr";
+  version = "1.4.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "XPixelGroup";
+    repo = "BasicSR";
+    rev = "v${version}";
+    hash = "sha256-UfwwwF0h0c5oPeGhj0w5Zw2edjPNoQWNG4pKHBwMU2I=";
+  };
```

Comment:
`CodeFormer` vendors a copy which contains some extra files. They're going to collide and I suspect we might want to just keep the copy?

Reply:
Where do these extra files originate from? Really a fan of unvendoring this garbage as much as possible.

Reply:
H'm maybe the vendored copy is just older, not sure if they actually patched it: https://gist.github.com/SomeoneSerge/595cf65e6acb86966e01b9f71363d3cd